### PR TITLE
Correctly detect change changes in repo

### PR DESCRIPTION
**Pull request by tamarakaufler**

### CODE REVIEW

**Comment by squaremo** [comment]
Some explanation in the commit message, please.

**Comment by tamarakaufler** [comment]
Commit message updated to provide more detail

**Comment by squaremo** [comment]
> Bug fix for chart sync 

What was wrong with the chart sync? What (briefly) was changed to fix it? Please phrase in imperative terms; e.g, as if completing the sentence "When applied, this commit will ..."

>
> Uses "git diff -r" command to find new commits

Why? What was wrong with what it was doing before?

> Removes now redundant cloning of the repo
> Minor improvements (variable renaming, debug cleanup)

(These ones are fine as bullet points.)

File: Gopkg.lock [hunk 1]
```diff
@@ -465,6 +465,6 @@
 [solve-meta]
   analyzer-name = "dep"
   analyzer-version = 1
-  inputs-digest = "3629212a518505ac90fb570403f2993c2b16257cff30ecc50363cfadd1c686e4"
+  inputs-digest = "db30bb7a0e5c03ea865378d78c65b2cadbeea1efd3e96f68afae43ebc11cc1a2"
```
Comment (squaremo) [inline Gopkg.lock]: Why is this here?

File: cmd/helm-operator/main.go [hunk 2]
```diff
@@ -228,33 +228,14 @@ func main() {
...
-	checkoutChCurr := git.NewCheckout(log.With(logger, "component", "git"), gitRemoteConfigCh, gitAuth)
-	defer checkoutChCurr.Cleanup()
+	checkoutCh := git.NewCheckout(log.With(logger, "component", "git"), gitRemoteConfigCh, gitAuth)
+	defer checkoutCh.Cleanup()
 
 	// If cloning not immediately possible, we wait until it is -----------------------------
 	for {
 		mainLogger.Log("info", "Cloning repo ...")
 		ctx, cancel := context.WithTimeout(context.Background(), git.DefaultCloneTimeout)
-		err = checkoutChCurr.Clone(ctx, git.ChartsChangesCloneCurr)
```
Comment (squaremo) [inline cmd/helm-operator/main.go]: Let's just never mention this change again.

File: integrations/helm/chartsync/chartsync.go [hunk 3]
```diff
@@ -98,85 +96,76 @@ func (chs *ChartChangeSync) Run(stopCh <-chan struct{}, errc chan error, wg *syn
 			select {
 			// ------------------------------------------------------------------------------------
 			case <-ticker.C:
-				fmt.Printf("\n\t... TICK at %s\n\n", time.Now().String())
+				fmt.Printf("\n\t... CHARTSYNC at %s\n\n", time.Now().String())
```
Comment (squaremo) [inline integrations/helm/chartsync/chartsync.go]: Is this debug output?

Comment (squaremo) [reply integrations/helm/chartsync/chartsync.go]: _passim_

File: Gopkg.lock (hunk 1 above)
Comment (tamarakaufler) [reply Gopkg.lock]: I vaguely remember that this changed after doing dep ensure, so I added it.

File: integrations/helm/chartsync/chartsync.go (hunk 3 above)
Comment (tamarakaufler) [reply integrations/helm/chartsync/chartsync.go]: Yes, it is. I prefer to leave it here until we get to a stable phase.
